lib/src/room.dart

```diff
@@ -666,6 +673,10 @@ class KurentoRoom extends ChangeNotifier implements Room {
   void _handleConnectionState(int trackId, RTCPeerConnectionState state) {
     // REVIEW: Should the room expose the connection state (e.g. `isAgentStreamConnected`, `isExplorerStreamConnected`)?
     _log.info('connection state changed track_id=$trackId state=$state');
+    if (RTCPeerConnectionState.RTCPeerConnectionStateFailed == state) {
```

Comment:
If the connection is in the `failed` state, why do we remove it and call `onConnectionFailed` instead of calling `restartIce` like we do when connectivity changes?

Reply:
Good idea! I'll add a retry mechanism.

Reply:
I have checked a couple of things here and `restartIce` would have to be called on the `RTCPeerConnection` object -- which has failed. I could recreate the full "peerConnection", but this would not be a trivial task ask the SFUConnection's creation is async and will succeed or fails way after the `createPeerConnection` (in `SFUConnection.connect`) returns we our state at that time can vary.

All this combined with the fact that this error will happen pretty shortly after the Explorer attempts to call Aira seems to indicate this would not be too much to ask our Explorer to retry.

All that would be missing is the added simplicity to be returned to the "call with message" page for an easy retry without loosing message or file to share with the agent. This is what I opted to do to remedy to that issue and you should be able to see an additional commit to PR1080 addressing this.